app/routes.py

```diff
@@ -1,2 +1,333 @@
+    header = {'Authorization': os.environ.get('BOT')}
+    response = requests.post('https://slack.com/api/chat.postMessage', params = query_path, headers = header)
+    return response.json()
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def update_complete(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+
+    if task == None:
+        return jsonify(None), 404
+
+    task.completed_at = datetime.now()
+    db.session.commit()
+
+    slack_bot(task.title)
+
+    task_dict = {}
+    task_dict = {
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at is not None
+            }
+        }
+    
+    return jsonify(task_dict), 200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def update_incomplete(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+
+    if task == None:
+        return jsonify(None), 404
+    
+    task.completed_at = None
+
+    db.session.commit()
+
+    task_dict = {}
+    task_dict = {
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at is not None
+            }
+        }
+    
+    return jsonify(task_dict), 200
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+
+    new_goal = Goal(
+        title=request_body["title"]
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response_body = {
+        "goal": {
+            "id": new_goal.goal_id,
+            "title": new_goal.title
+        }
+    }
+
+    return jsonify(response_body), 201
+
+@goals_bp.route("", methods=["GET"])
+def read_goals():
+    title_query = request.args.get("sort") 
+
+    if title_query == "asc":
+        goals = Goal.query.order_by(Goal.title.asc())
+    elif title_query == "desc":
+        goals = Goal.query.order_by(Goal.title.desc())
+    else:
+        goals = Goal.query.all() 
+
+    goal_responses = []
+    for goal in goals:
+        goal_responses.append(
+                {
+                "id": goal.goal_id,
+                "title": goal.title
+                }
+            )
+    
+    return jsonify(goal_responses), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def read_goal(goal_id):
+    goal_id = int(goal_id)
+    goal = Goal.query.get(goal_id)
+    
+    goal_dict = {}
+    if goal == None:
+        return jsonify(None), 404
+    else: 
+        goal_dict = {
+            "goal": {
+                "id": goal.goal_id,
+                "title": goal.title
+        }
+    }
```

Comment:
This logic doesn't need to be in the `else` logic. And it would also be helpful for the `Goal` model to have a dictionary helper method.